api/network/floating_ips_api.go

```diff
@@ -0,0 +1,180 @@
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/ingrammicro/concerto/api/types"
+	"github.com/ingrammicro/concerto/utils"
+)
+
+// FloatingIPService manages FloatingIP operations
+type FloatingIPService struct {
+	concertoService utils.ConcertoService
+}
+
+// NewFloatingIPService returns a Concerto FloatingIP service
+func NewFloatingIPService(concertoService utils.ConcertoService) (*FloatingIPService, error) {
+	if concertoService == nil {
+		return nil, fmt.Errorf("must initialize ConcertoService before using it")
+	}
+
+	return &FloatingIPService{
+		concertoService: concertoService,
+	}, nil
+}
+
+// GetFloatingIPList returns the list of FloatingIPs as an array of FloatingIP
+func (dm *FloatingIPService) GetFloatingIPList(serverID string) (floatingIPs []*types.FloatingIP, err error) {
+	log.Debug("GetFloatingIPList")
+
+	path := "/network/floating_ips"
+	if serverID != "" {
+		path = strings.Join([]string{path, "?server_id=", serverID}, "")
```

Comment:
This may work but is not officially part of the public API, you should use /cloud/servers/:server_id/floating_ips instead

Reply:
Done!